### ISSUE TITLE
`object-curly-newline`: More customizable wrapping behavior

### ISSUE THREAD

**Issue by reteps** [eslint-stylistic/eslint-stylistic#914]
### Clear and concise description of the problem

I use eslint-stylistic alongside prettier, with the following config:

```ts
'@stylistic/object-curly-newline': [
        'error',
        {
          ObjectExpression: {
            minProperties: 2,
            multiline: true,
          },
        },
      ],
```

- If an object has two or more properties, it should be multiline
- If an object has one property, it should be on a single line

For example:

good:

```ts
      foo: { bar: 'baz' },
      foo2: { bar: ['baz'] },
```

bad:
```ts
foo3: {
bar: 'baz'
}
```

However, this breaks down when you have some really long field:

```ts
object: {
        property: 'this property is so super duper long that prettier wants to break this object up into multiple lines',
},
```

<img width="1416" height="378" alt="Image" src="https://github.com/user-attachments/assets/b2be2389-0d04-4aff-9711-b96e7aab643c" />

### Suggested solution

This seems tricky to solve, as any solution that would work with prettier here would need some context as to:

- The indent level of the object (e.g. how many spaces before `object: ` started
- The contents of the object itself

I would propose an option that allows for a user-provided function for checking if the contents should be ignored. The entire object body is passed in as a string to the function.

```ts
ignored: ({ startPosition, contents }) => {
  // any check
  return (contents.length >= 80);
}
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
Hi, sorry for the late reply.
ESLint's [rule system](https://eslint.org/docs/latest/extend/custom-rules#options-schemas) is built around JSON schemas, which don't support function-type values.
This means we're unable to implement function-based options at this time.

**Comment by reteps** [comment]
Hi @9romise , I thought about the behavior I wanted some more:

```
multiline: {
          oneOf: [
            {
              type: 'boolean',
            },
            {
              type: 'object',
              properties: {
                breakLength: {
                  type: 'integer',
                  minimum: 0,
                },
              },
              additionalProperties: false,
              required: ['breakLength'],
            },
          ],
        },
```

- `"multiline": { "breakLength": 80 }` is equivalent to `"multiline": true` but modifies the behavior of the `minProperties` option. If an object contained linebreaks and the fixed version of the object without linebreaks has a line length greater than `breakLength`, then the error is ignored.

I wasn't quite sure how to implement this however. For example:

```ts
let d = {
  foo: {
    bar: "some-super-duper-long-string-xxx-yyy-zzz-aaa-bbb-ccc-ddd-eee",
  }
};
```

The prototype I have right now errors, and wants to convert it to:

```ts
let d = { foo: {
            bar: "some-super-duper-long-string-xxx-yyy-zzz-aaa-bbb-ccc-ddd-eee",
          }
        }
```

I don't see how to implement this without walking the whole object to compute the estimated line length of the fixed version.

Which seems to require knowledge of how long the fixed child nodes would be.

**Comment by 9romise** [comment]
Sorry, I think this option is too specific and unrelated to the rule.
For niche requirements, we'd suggest you write your own rule to fix that
